data/pools/transactionPool.go

```diff
@@ -584,11 +607,21 @@ func (pool *TransactionPool) addToPendingBlockEvaluatorOnce(txgroup []transactio
 
 	if recomputing {
 		transactionGroupDuration := time.Now().Sub(transactionGroupStartsTime)
-		pool.assemblyMu.Lock()
-		defer pool.assemblyMu.Unlock()
-		if !pool.assemblyResults.ok {
-			if (err == ledger.ErrNoSpace || (pool.assemblyDeadline != time.Time{} && time.Now().After(pool.assemblyDeadline))) && (pool.assemblyRound <= pool.pendingBlockEvaluator.Round()) {
+		if !pool.assemblyResults.assemblyComplete {
+			pool.assemblyMu.Lock()
+			defer pool.assemblyMu.Unlock()
+			if pool.assemblyRound > pool.pendingBlockEvaluator.Round() {
+				// the block we're assembling now isn't the one the the AssembleBlock is waiting for. While it would be really cool
+				// to finish generating the block, it would also be pointless to spend time on it.
+				// we're going to set the ok and assemblyComplete to "true" so we can complete this loop asap
+				pool.assemblyResults.ok = true
+				pool.assemblyResults.assemblyComplete = true
+				stats.StopReason = telemetryspec.AssembleBlockAbandon
```

Comment:
is this used somewhere?

Reply:
I don't believe so; however, just for completeness, I figure that I should update the statistics so that they won't be left uninitialized while the `ok` variable was set.

Reply:
done.